porch/pkg/git/git.go

```diff
@@ -186,6 +189,35 @@ func (r *gitRepository) Close() error {
 	return nil
 }
 
+func (r *gitRepository) Version(ctx context.Context) (string, error) {
```

Comment:
How should we think about the version for the git repository calculated here? If I understand the code correctly it essentially hashes all the refs and uses that as the version. Would this handle commits that happens to existing refs (which I think primarily branches and tags) in git? In particular, would this version change if someone adds an additional commit on the main branch?

Reply:
Yes. This checks the existence of refs, plus the commit hash to which they point. A push to main updates the main head ref to point to that new commit. 

So this will change for any new or deleted tag, for a commit to any branch, for a new or deleted branch, or for a forced push, or for a tag move. AFAIK, there is no change you could make to a remote that this would not pick up. Maybe some sort of magic where you add a commit but somehow have no ref to it. But we would not care about such a change anyway (if it is even possible).

Reply:
Yeah, by using both the refs and the commit hash it should cover all scenarios I can think of.